test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -376,8 +379,14 @@ public void execute(ClientYamlTestExecutionContext executionContext) throws IOEx
             // This is really difficult to express just with features, so I will break it down into 2 parts: version check for v7,
             // and feature check for v8. This way the version check can be removed once we move to v9
             @UpdateForV9
-            var fixedInV7 = executionContext.esVersion().major == Version.V_7_17_0.major
-                && executionContext.esVersion().onOrAfter(Version.V_7_17_2);
+            var fixedInV7 = executionContext.nodesVersions()
+                .stream()
+                .map(ESRestTestCase::parseLegacyVersion)
+                .flatMap(Optional::stream)
+                .min(VersionId::compareTo)
+                .map(v -> v.major == Version.V_7_17_0.major && v.onOrAfter(Version.V_7_17_2))
```

Comment:
No strong preference, but prob easier to understand the intend using filter / isPresent instead 
```java
.filter(v -> v.major == Version.V_7_17_0.major && v.onOrAfter(Version.V_7_17_2))
.isPresent()
```